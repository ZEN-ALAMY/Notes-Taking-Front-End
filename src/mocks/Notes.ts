import { Note } from "../types/Note";

const MOCK_NOTE: Note = {
  title: "My Notes",
  date: "19-09-2022",
  relevance: "Not relevant",
};

const MOCK_NOTES = [MOCK_NOTE];

export { MOCK_NOTE, MOCK_NOTES };
