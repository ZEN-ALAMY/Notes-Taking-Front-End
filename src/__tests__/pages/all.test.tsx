import { render, waitFor } from "@testing-library/react";
import { MOCK_NOTES } from "../../mocks/Notes";
import All from "../../pages/all";

jest.mock("../../services/notes-taking.service", () => ({
  getAllNotes: () => MOCK_NOTES,
}));

describe("All notes tests", () => {
  it("should render the UI correctly", async () => {
    await waitFor(() => {
      expect(render(<All />).asFragment()).toMatchSnapshot();
    });
  });
});
