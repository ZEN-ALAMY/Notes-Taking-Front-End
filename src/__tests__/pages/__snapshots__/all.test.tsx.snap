// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All notes tests should render the UI correctly 1`] = `
<DocumentFragment>
  <div
    class="bg-green-500 sm:bg-red-500 xl:bg-blue-500 h-[21%] flex flex-col  items-center justify-evenly text-black font-mono font-black "
  >
    <h1
      class="text-3xl sm:text-4xl xl:text-5xl font-extrabold"
    >
      NOTES TAKING APP
    </h1>
    <div
      class="flex flex-row gap-x-4 xl:gap-x-8 sm:gap-x-6"
    >
      <button
        class="bg-slate-100 rounded-xl border-none xl:w-[100px] sm:w-[80px] w-[60px]"
      >
        <span
          class="text-sm sm:text-lg xl:text-2xl"
        >
          HOME
        </span>
      </button>
      <button
        class="bg-slate-100 rounded-xl border-none xl:w-[100px] sm:w-[80px] w-[60px]"
      >
        <span
          class="text-sm sm:text-lg xl:text-2xl"
        >
          ALL
        </span>
      </button>
      <button
        class="bg-slate-100 rounded-xl border-none xl:w-[100px] sm:w-[80px] w-[60px]"
      >
        <span
          class="text-sm sm:text-lg xl:text-2xl"
        >
          ADD
        </span>
      </button>
      <button
        class="bg-slate-100 rounded-xl border-none xl:w-[100px] sm:w-[80px] w-[60px]"
      >
        <span
          class="text-sm sm:text-lg xl:text-2xl"
        >
          REMOVE
        </span>
      </button>
    </div>
  </div>
  <div
    class="h-screen"
  />
</DocumentFragment>
`;
